romancal/skycell/skymap.py

```diff
@@ -0,0 +1,635 @@
+        """
+        Retrieve a sky cell from the sky map by its name (see handbook [1] for explanation).
+
+        Parameters
+        ----------
+        name : str
+            Name of a sky cell, for instance `315p86x50y75`.
+        skymap: SkyMap
+            sky map instance (defaults to global SKYMAP)
+
+        References
+        ----------
+        .. [1] `Skymap Tessellation <https://roman-docs.stsci.edu/data-handbook-home/wfi-data-format/skymap-tessellation>`_
+        """
+        if not re.match(r"\d{3}\w\d{2}x\d{2}y\d{2}", name):
+            raise ValueError(f"invalid skycell name {name}")
+
+        if skymap is None:
+            skymap = SKYMAP
+
+        indices = np.where(skymap.model.skycells["name"] == name)[0]
+        if len(indices) > 0:
+            return SkyCell(indices[0], skymap=skymap)
+        else:
+            raise KeyError(
+                f"skycell with name '{name}' does not exist in the currently-loaded sky map"
+            )
+
+    @classmethod
+    def from_data(cls, data: np.void, skymap: "SkyMap" = None) -> "SkyCell":
+        """
+        build an index-less sky cell instance from a data array
+
+        Parameters
+        ----------
+        data : numpy.void
+            array with sky cell parameters (see schema)
+        skymap: SkyMap
+            sky map instance (defaults to global SKYMAP)
+        """
+        instance = cls(index=None, skymap=skymap)
+        instance._data = data
+        return instance
+
+    @classmethod
+    def from_center_and_coordinates(
+        cls,
+        projregion_center: tuple[float, float],
+        skycell_coordinates: tuple[int, int],
+        skymap: "SkyMap" = None,
+    ) -> "SkyCell":
+        """
+        Retrieve a sky cell from the sky map using
+        - the center of its containing projection region
+        - the ordinal XY coordinates of the sky cell from the projection region origin (number of sky cells away from the center)
+
+        (see handbook [1] for further explanation)
+
+        Parameters
+        ----------
+        projregion_center : tuple[float, float]
+            center coordinates of its containing projection region in right ascension and declination
+        skycell_coordinates : tuple[int, int]
+            XY location of the sky cell within its projection region, in units of ordinal sky cells from the center
+        skymap: SkyMap
+            sky map instance (defaults to global SKYMAP)
+
+        References
+        ----------
+        .. [1] `Skymap Tessellation <https://roman-docs.stsci.edu/data-handbook-home/wfi-data-format/skymap-tessellation>`_
+        """
+        return cls.from_name(
+            f"r{round(projregion_center[0]):03}d{'p' if projregion_center[1] >= 0 else 'm'}{round(projregion_center[1]):02}x{'p' if skycell_coordinates[1] >= 0 else 'm'}{skycell_coordinates[1]:02}y{'p' if skycell_coordinates[1] >= 0 else 'm'}{skycell_coordinates[1]:02}",
+            skymap=skymap,
```

Comment:
This looks like it uses the old style names rather than the new style names?  (i.e., it has names like r270dp66x01y01 rather than 270p66x01y01).  Is that intentional?  Do you also need to do some absolute value-ing on projregion_center[1] and skycell_coordinates[1], so that you get something like m01 rather than m-1 or something?  I think also the 'x' bit of the skycell_coordinates needs to reference skycell_coordinates[0] rather than skycell_coordinates[1].  

But maybe this function just isn't used?  It looks broken to me.